tests/conftest.py

```diff
@@ -285,3 +191,202 @@ def project_main_module() -> types.ModuleType:
+    RockcraftServiceFactory.register("image", services.RockcraftImageService)
+
+    factory = RockcraftServiceFactory(app=APP_METADATA)
+
+    factory.update_kwargs("project", project_dir=in_project_path)
+    factory.update_kwargs("provider", work_dir=in_project_path)
+    factory.update_kwargs(
+        "image", project_dir=in_project_path, work_dir=in_project_path
+    )
+    factory.update_kwargs(
+        "lifecycle", work_dir=in_project_path, cache_dir=in_project_path / "cache"
+    )
+
+    # Mock out image info to avoid a umoci call requiring root privileges
+    mocker.patch.object(
+        services.RockcraftImageService,
+        "_create_image_info",
+        return_value=default_image_info,
+    )
+
+    return factory
+
+
+@pytest.fixture
+def configured_project(fake_services: ServiceFactory, fake_project_file) -> None:
+    project_service = fake_services.get("project")
+    project_service.configure(platform=None, build_for=None)
+
+
+@pytest.fixture
+def fake_app(fake_services: services.RockcraftServiceFactory) -> Rockcraft:
+    from rockcraft.cli import fill_command_groups
+
+    app = Rockcraft(app=APP_METADATA, services=fake_services)
+    fill_command_groups(app)
+
+    return app
+
+
+@pytest.fixture
+def fake_app_config(fake_app: Rockcraft) -> dict[str, Any]:
+    return fake_app.app_config
+
+
+@pytest.fixture
+def default_project_dict() -> dict[str, Any]:
+    return {
+        "name": "test-rock",
+        "version": "0.1",
+        "summary": "Rock on!",
+        "description": "Ramble off!",
+        "base": "ubuntu@24.04",
+        "platforms": {
+            "risky": {
+                "build-on": ["amd64", "arm64", "ppc64el", "riscv64", "s390x"],
+                "build-for": ["riscv64"],
+            }
+        },
+    }
+
+
+@pytest.fixture
+def fake_project_yaml(
+    request: pytest.FixtureRequest,
+    default_project_dict: dict[str, Any],
+    mocker,
+) -> str:
+    from craft_application.util import dump_yaml
+
+    project = default_project_dict
+
+    # Optional overrides. See the `project_keys` fixture for more info.
+    if "project_keys" in request.fixturenames:
+        project_keys = request.getfixturevalue("project_keys")
+
+        project |= project_keys
+    if "base" in request.fixturenames:
+        project["base"] = str(request.getfixturevalue("base"))
+
+    base = craft_platforms.DistroBase.from_str(cast(str, project.get("base")))
+    mocker.patch.object(
+        craft_platforms.DistroBase, "from_linux_distribution", return_value=base
+    )
+
+    return dump_yaml(project)
+
+
+@pytest.fixture
+def fake_project_file(in_project_path, fake_project_yaml) -> Path:
+    project_file = in_project_path / "rockcraft.yaml"
+    project_file.write_text(fake_project_yaml)
+
+    return project_file
+
+
+@pytest.fixture
+def project_keys(request: pytest.FixtureRequest) -> dict[str, Any]:
+    """Fixture to modify the default project as needed.
```

Comment:
👍🏾

Reply:
Hol'up really?

I ask because when @bepri and I were writing it I thought you would have the most/best ideas for improvements.

Context: I really want to add a fixture like this to craft-application's pytest plugin so we can have a uniform way to generate projects, and this was a first attempt. (The craft-application one would come with some serious docs about setting up your app's tests)

Reply:
I'm totally game for doing a craft-app design of such a fixture, but this is plenty good for what we need now (which is to merge this PR)